plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerBehaviour.java

```diff
@@ -374,19 +385,27 @@ public void onStart(Process process) {
   }
 
   /**
-   * An output listener that monitors for well-known key dev_appserver output and affects server
+   * An output listener that monitors for well-known key dev_appserver output and effects server
    * state changes.
    */
   public class DevAppServerOutputListener implements ProcessOutputLineListener {
-    // DevAppServer outputs the following for module-started and admin line (on one line):
+    // DevAppServer2 outputs the following for module-started and admin line (on one line):
     // <<HEADER>> Starting module "default" running at: http://localhost:8080
     // <<HEADER>> Starting admin server at: http://localhost:8000
     // where <<HEADER>> = INFO 2017-01-31 21:00:40,700 dispatcher.py:197]
-    private Pattern moduleStartedPattern = Pattern.compile(
+    
+    // devappserver2 patterns
+    private final Pattern moduleStartedPattern = Pattern.compile(
         "INFO .*Starting module \"(?<service>[^\"]+)\" running at: (?<url>http://.+:(?<port>[0-9]+))$");
-    private Pattern adminStartedPattern =
+    private final Pattern adminStartedPattern =
         Pattern.compile("INFO .*Starting admin server at: (?<url>http://.+:(?<port>[0-9]+))$");
 
+    // devappserver1 patterns
+    private final Pattern moduleRunningPattern = Pattern.compile(
+        "INFO: Module instance (?<service>\\w+) is running at (?<url>http://.+:(?<port>[0-9]+)/)$");
```

Comment:
I see `\w` is `[a-zA-Z_0-9]`, which does not include a hyphen. Not sure, but can a service name contain hyphens? Probably just safe to do `[^ ]+`. (Not tested.)